blocks/datasets/schemes.py

```diff
@@ -91,5 +92,22 @@ def __init__(self, num_examples, batch_size):
         self.batch_size = batch_size
 
     def get_request_iterator(self):
-        return (slice(x, min(self.num_examples, x + self.batch_size))
-                for x in range(0, self.num_examples, self.batch_size))
+        return SequentialIterator(self.num_examples, self.batch_size)
+
+
+class SequentialIterator(six.Iterator):
```

Comment:
No, it was never a list, the parentheses made it a generator, so it used just as much memory.